### DISCUSSION TITLE
Adding "Read Only"

### DISCUSSION THREAD

**Discussion by sigzero** [orbitalquark/textadept#85] [General]
I have implemented the following per the book:

```
-- Add a readonly flag for the command line
args.register('-r', '--readonly', 0, function()
  -- Make all opened buffers read-only.
  events.connect(events.FILE_OPENED, function()
    buffer.read_only = true
    ui.statusbar_text = 'READ ONLY'
  end)
  textadept.menu.menubar = nil -- hide the menubar
end, "Read-only mode")
```

I added the `ui.statusbar_text` option which works (meaning it shows the text in the statusbar) until I hit any key on the keyboard and then it goes away.

Is there a way to add `ui.statusbar_text = 'READ ONLY'` to the statusbar for any buffer that is open AND make it persistent?

**Comment by orbitalquark** [comment]
You can connect to `events.KEYPRESS`:

  events.connect(events.KEYPRESS, function()
    if buffer.read_only then ui.statusbar_text = 'READ ONLY' end
  end)

Textadept's default keypress event writes to the statusbar, so you can do the same.

**Reply by sigzero** [reply]
Awesome, thanks.
